src/Unused/CLI/Util.hs

```diff
@@ -28,15 +29,32 @@ withInterruptHandler body = do
     void $ installHandler keyboardSignal (Catch (handleInterrupt tid)) Nothing
     body
 
+withChildInterruptHandler :: ThreadId -> IO a -> IO a
+withChildInterruptHandler tid body = do
+    currentThread <- myThreadId
+    void $ installHandler keyboardSignal (Catch (handleChildInterrupt currentThread tid)) Nothing
+    body
```

Comment:
Do you need to accept `body` as an argument here? It looks like you could drop it, and then do:

``` haskell
installChildInterruptHandler tid
body
```

(from the reference location)